### PR TITLE
catch axios and express vlaidator errors properly

### LINKED ISSUE #954: Catch error.response.data in UI to display nicer error messages
As it was done in https://github.com/globaldothealth/list/pull/950

### PR DESCRIPTION
**Pull request by attwad**
As I did in #950 previously but even simpler this time using ||.

Example of error before:
image

(styling is weird btw for large errors)

Example of same error now:

image

Fixes #954

### CODE REVIEW

File: verification/curator-service/ui/src/components/App.tsx [hunk 1]
```diff
@@ -308,7 +308,6 @@ export default function App(): JSX.Element {
             })
             .catch((e) => {
                 setUser({ _id: '', name: '', email: '', roles: [] });
-                console.error(e);
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/App.tsx:311]: Why remove?

**Review by allysonjp715 — APPROVED** [review]
Thank you!!

File: verification/curator-service/ui/src/components/App.tsx (hunk 1 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/App.tsx:311]: This is going to print a console error for all logged out users, not super useful imo

image

**Comment by attwad** [comment]
Okay it turns out express validator errors have to be caught via a middleware for proper formatting (it's in their docs, don't know why we didn't follow those). So I did that.
And I made ALL our errors match the same principle that is the json output of errors should contain a message field otherwise the stringified version of the error output is used (catch-all in case the server doesn't follow that best practice).

message field is part of all errors in javascript so it actually makes the code simpler in a few places.

Now we get nice errors on 404s for example:
Before:
image

Now:
Screenshot 2020-09-08 at 11 21 28

**Review by allysonjp715 — APPROVED** [review]

**Pull request merged**